draft-ietf-masque-connect-udp-listen.md

```diff
@@ -154,10 +178,114 @@ UDP Payload:
 : The unmodified UDP Payload of this proxied UDP packet (referred to as "data
 octets" in {{UDP}}).
 
-# The Connect-UDP-Bind Header Field {#hdr}
+# Context ID {#contextid}
+
+This extension leverages context IDs (see {{Section 4 of CONNECT-UDP}}) to compress the target IP address and port when encoding datagrams on the wire. Either endpoint can register a context ID and the IP/ports it's associated with by sending a COMPRESSION_ASSIGN capsule to its peer. The peer will then echo that capsule to indicate it's received it. From then on, both endpoints are aware of the context ID and can send compressed datagrams. Later, any endpoint can decide to close the compression context by sending a COMPRESSION_CLOSE capsule.
+
+The context ID 0 was reserved by unextended connect-udp and is not used by this extension.
+Once an endpoint has ascertained that the peer supports this extension, the endpoint MUST NOT
+send any datagrams with context ID set to 0, and MUST drop any received datagrams with
+context ID set to 0.
+
+An even context ID MUST be used by clients to allocate uncompressed or compressed
+contexts, and an odd context ID MUST be used by the proxy to allocate compressed
+context IDs.They MAY pre-emptively use Context IDs not yet acknowledged by the other party, knowing that those packets MAY be lost since the COMPRESSION_ASSIGN request receiving proxy
```

Comment:
nit: this requirement comes from RFC 9298, so we don't need a new MUST. I would instead phrase it as a reference:

```suggestion
As mandated in {{Section 4 of CONNECT-UDP}}, clients will allocate even context IDs
while proxies will allocate odd ones.
They MAY pre-emptively use Context IDs not yet acknowledged by the other party, knowing that those packets MAY be lost since the COMPRESSION_ASSIGN request receiving proxy
```